Data/AuthenticationClient/Sources/Persistence.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  Persistence.swift
+//  QuranEngine
+//
+//  Created by Mohannad Hassan on 28/12/2024.
+//
+
+import Foundation
+import VLogging
+
+enum PersistenceError: Error {
+    case persistenceFailed
+    case retrievalFailed
+}
+
+/// An abstraction for secure persistence of the authentication state.
+protocol Persistence {
+    func persist(state: Data) throws
+
+    func retrieve() throws -> Data?
```

Comment:
To make this generic:

1. I think we need to make this keyed persistence to allow saving and retrieving different kind of information.
2. Can we move it under Core/ as a separate target please?